source/common/network/resolver_impl.cc

```diff
@@ -67,7 +67,17 @@ resolveProtoSocketAddress(const envoy::api::v2::core::SocketAddress& socket_addr
   if (resolver == nullptr) {
     throw EnvoyException(fmt::format("Unknown address resolver: {}", resolver_name));
   }
-  return resolver->resolve(socket_address);
+
+  try {
+    return resolver->resolve(socket_address);
+  } catch (EnvoyException& e) {
+    // resolver_name.empty() means resolver tries to resolve using IP address.
+    if (resolver_name.empty()) {
+      throw EnvoyException(fmt::format(
+          "{}. Please try to use 'STRICT_DNS' or 'LOGICAL_DNS' cluster type instead", e.what()));
```

Comment:
Not really sure if this is correct, should we just tell the user to set the `resolver_name` and define another cluster entry with ip address? (seems confusing while I'm typing this).

Reply:
I think the tricky part about an error here is that, in principal, there could be a named resolver that did DNS (or other hostname) lookups. No such resolver ships with Envoy, though. Setting the cluster type bypasses this resolver altogether, which is the currently the only stock way to use DNS names, AFAIK.

Perhaps the addition to the error should be "Consider setting resolver_name or setting cluster type to 'STRICT_DNS' or 'LOGICAL_DNS'."

Reply:
I wonder whether this type of error checking would be better placed more at the source? Like in the static cluster and the EDS cluster directly? Thoughts?

Reply:
@zuercher @mattklein123 sorry missed your comments, I tried to pass the `cluster_type` down, hence we will be sure about the current cluster type. What do you think?